src/transformState.js

```diff
@@ -6,6 +6,26 @@
  */
 function transformState(state, actions) {
   // write code here
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'removeProperties':
+        for (const key1 of obj.keysToRemove) {
+          delete state[key1];
+        }
+        break;
+
+      case 'addProperties':
+        for (const key2 in obj.extraData) {
```

Comment:
You can use Object.assign() instead of this loop, where add obj.extraData into state object.